### PR TITLE
Add devise for auth

### PR DESCRIPTION
**Pull request by AliRezaBmeDu**
In this branch, we have changed the following things.

- Remove current_user method in ApplicationController, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- Can register a new user.

    1. User logs in with a combination of email and password.
    2. Hashed password should be stored in the database.
    3. Ask for confirmation of email.
    4. Can reset password.
    
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        <div class="text-center">
+          <%= f.submit "Log in", class: "btn btn-primary" %>
+        </div>
+      <% end %>
+
+      <%= render "devise/shared/links" %>
+    </div>
+  </div>
+</div>
```
Comment (rloterh) [inline app/views/devise/sessions/new.html.erb:32]: - [ ] Nice job on configuring devise authentication parameters correctly and the awesome login layout 👌This activity, however, requires the modification of the views in _devise_ for the registration also matches the styling of the [provided wireframe(s)](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md). Kindly ensure that the form (inputs and buttons) stretches across the screen and is centered accordingly for the _Signup_ and _Login_ pages, as seen in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md):

**Your Signup view:**        |  **Your login view:**        |  **Wireframe login view (Expected):**
:-------------------------:|:-------------------------:|:-------------------------:
Home Page | Home Page | Home Page

**Review by rloterh — COMMENTED** [review]
Checked!

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (rloterh) [inline app/views/devise/sessions/new.html.erb]: - [x] Nice job on configuring devise authentication parameters correctly and the awesome login layout 👌This activity, however, requires the modification of the views in _devise_ for the registration also matches the styling of the [provided wireframe(s)](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md). Kindly ensure that the form (inputs and buttons) stretches across the screen and is centered accordingly for the _Signup_ and _Login_ pages, as seen in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md):

**Your Signup view:**        |  **Your login view:**        |  **Wireframe login view (Expected):**
:-------------------------:|:-------------------------:|:-------------------------:
Home Page | Home Page | Home Page

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ali, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 ⛪ 🚜 🏃‍♀️ 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Added required Authentication parameters!
- Devise Authentication works as intended!
- Fantastic UI!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AliRezaBmeDu and @SaveryIV,

✔️ Integration of Devise for authentication of users

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**